scripts/irods/test/test_rule_engine_plugin_passthrough.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import print_function
+import os
+import sys
+import mmap
+
+if sys.version_info < (2, 7):
+    import unittest2 as unittest
+else:
+    import unittest
+
+from . import session
+from .. import test
+from .. import lib
+from .. import paths
+from ..configuration import IrodsConfig
+
+class Test_Rule_Engine_Plugin_Passthrough(session.make_sessions_mixin([('otherrods', 'rods')], []), unittest.TestCase):
+
+    def setUp(self):
+        super(Test_Rule_Engine_Plugin_Passthrough, self).setUp()
+        self.admin = self.admin_sessions[0]
+
+    def tearDown(self):
+        super(Test_Rule_Engine_Plugin_Passthrough, self).tearDown()
+
+    @unittest.skipIf(test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing")
+    def test_repf_continuation_using_passthrough_rep__issues_4147_4148_4179(self):
+        config = IrodsConfig()
+
+        with lib.file_backed_up(config.server_config_path):
+            with lib.file_backed_up(config.client_environment_path):
+                pep_name = 'pep_database_open_pre'
+
+                # Enable the Passthrough REP and set the log level to debug.
+                # The Passthrough REP will be triggered before the native REP.
+                config.server_config['plugin_configuration']['rule_engines'].insert(0, {
+                    'instance_name': 'irods_rule_engine_plugin-passthrough-instance',
+                    'plugin_name': 'irods_rule_engine_plugin-passthrough',
+                    'plugin_specific_configuration': {
+                        'return_codes_for_peps': [
+                            {
+                                'regex': '^' + pep_name + '$',
+                                'code': 5000000
```

Comment:
can this be the NAME rather than the number?

Reply:
Sure, i'll give it a name.